packages/grafana-ui/src/components/Logs/LogDetails.tsx

```diff
@@ -9,33 +9,104 @@ import { getLogRowStyles } from './getLogRowStyles';
 //Components
 import { LogDetailsRow } from './LogDetailsRow';
 
+type FieldDef = {
+  key: string;
+  value: string;
+  links?: string[];
+  isDerived?: boolean;
+  fieldIndex?: number;
+};
+
 export interface Props extends Themeable {
   row: LogRowModel;
   getRows: () => LogRowModel[];
   onClickFilterLabel?: (key: string, value: string) => void;
   onClickFilterOutLabel?: (key: string, value: string) => void;
+  getFieldLinks?: (field: Field, rowIndex: number) => Array<LinkModel<Field>>;
 }
 
 class UnThemedLogDetails extends PureComponent<Props> {
   parseMessage = memoizeOne(
-    (rowEntry): { parsedFields: string[]; parser?: LogsParser } => {
+    (rowEntry): { fields: FieldDef[]; parser?: LogsParser } => {
       const parser = getParser(rowEntry);
       if (!parser) {
-        return { parsedFields: [] };
+        return { fields: [] };
       }
       // Use parser to highlight detected fields
       const parsedFields = parser.getFields(rowEntry);
-      return { parsedFields, parser };
+      const fields = parsedFields.map(field => {
+        const key = parser.getLabelFromField(field);
+        const value = parser.getValueFromField(field);
+        return { key, value };
+      });
+
+      return { fields, parser };
+    }
+  );
+
+  getDerivedFields = memoizeOne(
```

Comment:
named getDataFrameFields is maybe more descriptive as these can be derived or just part of the query result as additional fields